official/vision/beta/projects/basnet/modeling/basnet_encoder.py

```diff
@@ -0,0 +1,212 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""BASNet Encoder
```

Comment:
In general, I feel that class BASNet_Encoder() and class BASNet_decoder() can both be put into the basnet_model.py instead of putting in their separate .py. It is just my opinion.